### PR TITLE
Metrics basic deprecated validators removed in favor of pattern matching

### PR DESCRIPTION
**Pull request by dirrao**
Metrics basic deprecated validators (AllowListValidator and BlockListValidator) removed in favor of pattern matching

### CODE REVIEW

File: airflow/metrics/validators.py
Comment (ferruzzi) [inline airflow/metrics/validators.py]:
```suggestion
       "allow": PatternAllowListValidator, 
       "block": PatternBlockListValidator,
```

I don't see a reason to keep the "pattern" level anymore.

File: airflow/metrics/validators.py
Comment (ferruzzi) [inline airflow/metrics/validators.py]: If the above is accepted then also:

```suggestion
    return validators[list_type](metric_lists[list_type])
```

File: tests/core/test_stats.py
Comment (ferruzzi) [inline tests/core/test_stats.py:360]: Did we miss a case here where both allow and block are provided and we assert that allow is being used?

File: tests/core/test_stats.py
Comment (ferruzzi) [inline tests/core/test_stats.py]: Um akshully, that should be `baz` not `bax` :P   That's my typo.

**Review by ferruzzi — CHANGES_REQUESTED** [review]
I am so happy to see these deprecation PRs actually removing stuff!

Left an easy change request/suggestion, otherwise LGTM.

Those unit tests were somewhat convoluted, nice work sorting through them.

File: tests/core/test_stats.py
Comment (ferruzzi) [reply tests/core/test_stats.py]: I don't expect you to fix it, I just noticed it while reviewing your changes, feel free to resolve this.

Comment (dirrao) [reply tests/core/test_stats.py]: Updated.

File: airflow/metrics/validators.py
Comment (dirrao) [reply airflow/metrics/validators.py]: done

File: tests/core/test_stats.py
Comment (ferruzzi) [inline tests/core/test_stats.py:392]: LOL   Thanks

File: airflow/metrics/validators.py
Comment (ferruzzi) [inline airflow/metrics/validators.py:110]: Non-blocking:   Seeing this now, after the changes you made, I wonder if there is a better way to handle these two dicts.  

If you want to call this out of scope for this PR, I won't push it, but maybe consider changing this up.

With those two  getting simplified so much, there are far fewer permutations here and this entire method could be simplified down to something closer to ((untested code))

```
def get_validator():
    metric_allow_list = conf.get("metrics", "metrics_allow_list", fallback=None)
    metric_block_list = conf.get("metrics", "metrics_block_list", fallback=None)

    if metric_allow_list and metric_block_list:
        log.warning("Ignoring metrics_block_list as both metrics_allow_list and metrics_block_list have been set.")
    
    if metric_allow_list or DEFAULT_VALIDATOR_TYPE == "allow":
        return PatternAllowListValidator(metric_allow_list)
    if metric_block_list or DEFAULT_VALIDATOR_TYPE == "block":
        return PatternBlockListValidator(metric_block_list)
```

**Review by ferruzzi — APPROVED** [review]

File: airflow/metrics/validators.py
Comment (ferruzzi) [reply airflow/metrics/validators.py:110]: Or skip the DEFAULT_VALIDATOR_TYPE entirely and do something like

```
def get_validator():
    metric_allow_list = conf.get("metrics", "metrics_allow_list", fallback=None)
    metric_block_list = conf.get("metrics", "metrics_block_list", fallback=None)

    if metric_allow_list and metric_block_list:
        # If both are set, warn
        log.warning("Ignoring metrics_block_list as both metrics_allow_list and metrics_block_list have been set.")
    elif metric_block_list:
       # If not both but block is set, use that
       return PatternBlockListValidator(metric_block_list)

    # if none set, both are set, or only only allow is set then use allow
    return PatternAllowListValidator(metric_allow_list)        
```        


But maybe that's more confusing and the more verbose way is easier to understand.    I don't know, I'll leave it up to you unless someone else chimes in with stronger opinions.

Comment (dirrao) [reply airflow/metrics/validators.py:110]: Sure. I am planning to consider this code optimization in a separate MR.

Comment (ferruzzi) [reply airflow/metrics/validators.py:110]: Dirrao:  I can live with that.  I just ask that you leave this conversation open for others to chime in.


Committer:   If this PR has two approvals and this is the only conversation blocking merge, feel free to resolve it and merge.

**Comment by dirrao** [comment]
@potiuk / @eladkal 
I'm awaiting final approval to proceed with the merge. Could you please take a look when you get a chance?

**Review by potiuk — APPROVED** [review]

**Comment by dirrao** [comment]
> Needs to resolve conflict now.

Done.

**Comment by dirrao** [comment]
@eladkal 
I'm waiting for the final review and merge.

**Pull request merged**
